horace_core/sqw/@Experiment/Experiment.m

```diff
@@ -221,14 +187,45 @@
             % property, which inform about the behaviour of runid map
             obj.runid_recalculated_ = logical(val);
         end
-        %------------------------------------------------------------------
-        function nr = get.n_runs(obj)
-            nr = numel(obj.expdata_);
-            if nr == 1
-                if isempty(obj.expdata_(1))
-                    nr = 0;
-                end
+    end
+    %----------------------------------------------------------------------
+    % legacy instrument methods interface
+    %----------------------------------------------------------------------
+    methods
+        %Change fields in the experiment with corrections related to aligned
+        %crystal lattice parameters and orientation
+        obj=change_crystal(obj,alatt,angdeg,rlu_corr)
+
+        % add or reset instrument, related to the given experiment object
+        % array of instruments, or function, which defines the instrument
+        % with its possible parameters
+        obj = set_instrument(obj,instrument,varargin)
+        % set moderator pulse on every instrument contributing to the
+        % object
+        obj = set_mod_pulse(obj,pulse_model,pm_par)
+        % Return array of incident energies from all contributing runs
+        en = get_efix(obj);
+        function emode = get_emode(obj)
+            % Return array of instrument modes provided in all contributing runs
+            emode = arrayfun(@(x)x.emode,obj.expdata_,'UniformOutput',true);
+        end
+        function obj = set_efix_emode(obj,efix,emode)
+            % change efix and (optionally) emode in all experiment descriptions
+            % if emode is absent or described by any character string,
+            % the emode is kept unchanged
+            if nargin == 2
+                emode = '-keep_emode';
             end
```

Comment:
As mentioned before the intent is not the same and with new MATLAB syntax can break unexpectedly
```suggestion
            if ~exist('emode', 'var')
                emode = '-keep_emode';
            end
```

Reply:
as the algorithm stands, it may be emode but it may be empty